src/components/checkout/Checkout.js

```diff
@@ -1,93 +1,211 @@
 import React, { useState, useEffect } from "react";
-import { useProductsContext } from "../../context/products-context";
 import { useAuthContext } from "../../context/auth-context";
-import { v4 as uuidv4 } from "uuid";
 import "./Checkout.scss";
 
 const Checkout = () => {
-  const { products } = useProductsContext();
   const { user } = useAuthContext();
 
+  const [productsAddedToCart, setProductsAddedToCart] = useState([]);
+  const [subtotalOfProducts, setSubtotalOfProducts] = useState(0);
+  const [shippingCost, setShippingCost] = useState(9.9);
+
+  const getProductsArrInLocalStorage = () => {
+    if (user) {
+      setProductsAddedToCart(JSON.parse(localStorage.getItem(user.uid)));
+    } else {
+      setProductsAddedToCart([]);
+    }
+  };
+
+  const calculateSubtotalOfProducts = () => {
+    if (user && productsAddedToCart) {
+      const subtotalOfProducts = productsAddedToCart.reduce(
+        (acc, productObj) => {
+          return acc + productObj.subTotal;
+        },
+        0
+      );
+      setSubtotalOfProducts(Number(subtotalOfProducts));
+    }
+    console.log(typeof subtotalOfProducts);
+    console.log(typeof shippingCost);
+  };
+
+  useEffect(() => {
+    productsAddedToCart.length !== 0 && calculateSubtotalOfProducts();
+  }, [productsAddedToCart]);
+
+  useEffect(() => {
+    user && getProductsArrInLocalStorage();
+  }, [user]);
+
+  console.log(productsAddedToCart);
+  console.log(subtotalOfProducts);
```

Comment:
Good job! But can you put vanilla JS codes in useEffect or functions if possible? Because they will be executed many times.